File: 백준/Silver/10828. 스택/스택.py
import sys
from collections import deque

if __name__ == '__main__':
    N = int(sys.stdin.readline())
    stack = deque()

    for _ in range(N):
        inputs = sys.stdin.readline().split()

        if len(inputs) == 2:
            op, x = inputs[0], int(inputs[1])
            if op == "push":
                stack.append(x)
        else:
            op = inputs[0]
            if op == "pop":
                if len(stack) == 0:
                    print(-1)
                else:
                    print(stack.pop())
            elif op == "size":
                print(len(stack))
            elif op == "empty":
                if len(stack) == 0:
                    print(1)
                else:
                    print(0)
            elif op == "top":
                if len(stack) != 0:
                    n = stack.pop()
                    stack.append(n)
                    print(n)
                else:
                    print(-1)
            else:
                print(0)

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.Stack;

public class Main{
    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Stack<Integer> stack = new Stack<>();
        int T = Integer.parseInt(br.readLine());

        for (int i=0; i<T; i++) {
            String[] inputs = br.readLine().split(" ");

            String op = inputs[0];
            if (op.equals("push")) {
                stack.push(Integer.parseInt(inputs[1]));
            } else if (op.equals("pop")) {
                if (stack.isEmpty()) {
                    bw.write(-1+"\n");
                } else {
                    bw.write(stack.pop()+"\n");
                }
            } else if (op.equals("size")) {
                bw.write(stack.size()+"\n");
            } else if (op.equals("empty")) {
                int ans = (stack.empty() ? 1 : 0);
                bw.write(ans+"\n");
            } else if (op.equals("top")) {
                if (stack.isEmpty()) {
                    bw.write(-1+"\n");
                } else {
                    bw.write(stack.peek()+"\n");
                }
            } else {
                bw.write(0);
            }
        }
        bw.flush();
    }
}